recipes/qtxlsxwriter/all/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(qtxlsxwriter)
+
+find_package(qt REQUIRED CONFIG)
+
+set(QTXLSXWRITER_TARGET_NAME qtxlsxwriter)
+
+include(conan_qt_executables_variables)
+set(CMAKE_AUTOMOC ON)
+
+file(GLOB QTXLSXWRITER_HEADERS LIST_DIRECTORIES=false source_subfolder/src/xlsx/*.h)
+file(GLOB QTXLSXWRITER_SOURCES LIST_DIRECTORIES=false source_subfolder/src/xlsx/*.cpp)
+
+add_library(${QTXLSXWRITER_TARGET_NAME}
+	${QTXLSXWRITER_HEADERS}
+	${QTXLSXWRITER_SOURCES}
+)
+
+set(QT_VERSION "${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}.${QT_VERSION_PATCH}")
+target_include_directories(${QTXLSXWRITER_TARGET_NAME}
+	PRIVATE
+		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source_subfolder/src/xlsx>
+		# qtxlsxwriter uses qt private headers
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtCore/${QT_VERSION}>
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtGui/${QT_VERSION}>
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtGui/${QT_VERSION}/QtGui>
+	INTERFACE
+		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source_subfolder/src/xlsx>
+		$<INSTALL_INTERFACE:include>
+)
+
+target_compile_features(${QTXLSXWRITER_TARGET_NAME}
+	PRIVATE
+		cxx_range_for
+)
+
+# place qt::Gui before qt::Core (important for some toolchains)
+# since the current qt recipe provides an incorrect order of libraries
+target_link_libraries(${QTXLSXWRITER_TARGET_NAME}
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_RELEASE>:CONAN_LIB::qt_Qt5Gui_RELEASE>
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_DEBUG>:CONAN_LIB::qt_Qt5Gui_DEBUG>
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_RELWITHDEBINFO>:CONAN_LIB::qt_Qt5Gui_RELWITHDEBINFO>
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_MINSIZEREL>:CONAN_LIB::qt_Qt5Gui_MINSIZEREL>
+)
+
+target_link_libraries(${QTXLSXWRITER_TARGET_NAME}
+	qt::qt
+)
+
+get_target_property(target_type ${QTXLSXWRITER_TARGET_NAME} TYPE)
+if (target_type STREQUAL "STATIC_LIBRARY")
+	target_compile_definitions(${QTXLSXWRITER_TARGET_NAME}
+		PUBLIC
+			QTXLSX_STATIC
+	)
+endif()
+
+set_target_properties(${QTXLSXWRITER_TARGET_NAME} PROPERTIES
+	DEFINE_SYMBOL QT_BUILD_XLSX_LIB
+)
+
+set_property(TARGET ${QTXLSXWRITER_TARGET_NAME} PROPERTY PUBLIC_HEADER
```

Comment:
Could this be the header glob from above?

Reply:
The `GLOB` from above collects all headers from the directory, but this directory contains private headers as well (they have the `_p` postfix), so public headers are specified separately.

Reply:
It's hard to say, but should we use a glob here too so we don't have to change the list for each new version?

Reply:
This `CMakeLists` is not a part of `qtxlsxwriter` and most probably won't be maintained by the author of the library, so I think it's safer to glob the list of public headers. In this case we don't risk to miss addition of a new header to the list.